pkgs/by-name/ma/marmite/package.nix

```
@@ -0,0 +1,48 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  oniguruma,
+  stdenv,
+  darwin,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "marmite";
+  version = "0.2.3";
+
+  src = fetchFromGitHub {
+    owner = "rochacbruno";
+    repo = "marmite";
+    rev = version;
+    hash = "sha256-AblitYe7YDUc2Tg2P7j+wnOjMAhDtieDsbq6B6x+uMs=";
+  };
+
+  cargoHash = "sha256-u8sJS9hvIao+af7IA4vE0eUSx0xmI1Kb5NXyZBrOCIw=";
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs =
+    [
+      oniguruma
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.CoreFoundation
+      darwin.apple_sdk.frameworks.CoreServices
+    ];
```

This shouldn't be necessary nowadays with the new apple-sdk pattern
```suggestion
    ;
```